kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/TopicEncryptionST.java

```diff
@@ -0,0 +1,132 @@
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.strimzi.api.kafka.model.Kafka;
+
+import io.kroxylicious.systemtests.extensions.KroxyliciousExtension;
+import io.kroxylicious.systemtests.installation.kroxylicious.Kroxylicious;
+import io.kroxylicious.systemtests.installation.vault.Vault;
+import io.kroxylicious.systemtests.resources.vault.VaultTestKekManager;
+import io.kroxylicious.systemtests.steps.KafkaSteps;
+import io.kroxylicious.systemtests.steps.KroxyliciousSteps;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaNodePoolTemplates;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaTemplates;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ExtendWith(KroxyliciousExtension.class)
+class TopicEncryptionST extends AbstractST {
+    protected static final String BROKER_NODE_NAME = "kafka";
+    private static final Logger LOGGER = LoggerFactory.getLogger(TopicEncryptionST.class);
+    private static final String MESSAGE = "Hello-world";
+    private static Vault vaultOperator;
+    private static Kroxylicious kroxylicious;
+    private final String clusterName = "my-cluster";
+
+    @BeforeAll
+    void setUp() {
+        List<Pod> vaultPods = kubeClient().listPodsByPrefixInName(Constants.VAULT_DEFAULT_NAMESPACE, Constants.VAULT_SERVICE_NAME);
+        if (!vaultPods.isEmpty()) {
+            LOGGER.warn("Skipping vault deployment. It is already deployed!");
+        }
+        else {
+            vaultOperator = new Vault(Constants.VAULT_DEFAULT_NAMESPACE);
+            NamespaceUtils.createNamespaceWithWait(Constants.VAULT_DEFAULT_NAMESPACE);
+            vaultOperator.deploy();
+        }
+
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(Constants.KROXY_DEFAULT_NAMESPACE, clusterName);
+        if (!kafkaPods.isEmpty()) {
+            LOGGER.warn("Skipping kafka deployment. It is already deployed!");
+            return;
+        }
+        LOGGER.info("Deploying Kafka in {} namespace", Constants.KROXY_DEFAULT_NAMESPACE);
+
+        Kafka kafka = KafkaTemplates.kafkaPersistentWithKRaftAnnotations(Constants.KROXY_DEFAULT_NAMESPACE, clusterName, 3).build();
+
+        resourceManager.createResourceWithWait(
+                KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(BROKER_NODE_NAME, kafka, 3).build(),
+                kafka);
+    }
+
+    @AfterAll
+    void cleanUp() throws IOException {
+        if (vaultOperator != null) {
+            vaultOperator.delete();
+        }
+    }
+
+    @Test
+    void produceAndConsumeEncryptedMessage(String namespace) {
+        String topicName = "my-topic";
+        int numberOfMessages = 1;
+        String expectedMessage = MESSAGE + " - " + (numberOfMessages - 1);
+
+        var testKekManager = new VaultTestKekManager(Constants.VAULT_DEFAULT_NAMESPACE, Constants.VAULT_SERVICE_NAME + "-0");
```

Comment:
Ultimately, the system tests should be capable of testing any Kms, not just Vault.
Did you consider using the TestKmsFacadeFactory here?

The system tests would need its own implementation of the VaultTestKmsFacadeFactory  (a KubeVaultVaultTestKmsFacadeFactory) which would operate on the a vault instance within Kube (rather than within Docker).

Reply:
KubeVaultTestKmsFacadeFactory created